### ISSUE TITLE
Add function to print floats with thousands separators (for prices)

### ISSUE THREAD

**Issue by AeonFr** [gohugoio/hugo#5049]
I wanted to print a price in Hugo, using the native functions. I came across `printf`, witch lets you format a number with a fixed amount of decimals. So `85900` can be printed as `85900.00`.

_(I know this is not a forum or a tutorial, but I wanted to share how I managed to print the price so far in my site. At the bottom there's a TL;DR in case you want to get to the point)_

## How I managed so far

In Spanish, decimal values go with a comma, not a dot, but I could easily change the dot and replace it entirely (in my case I prefered to use a `<sup>` tag for decimals, so I simply replaced the dot with the tag and wrapped all in the `safeHTML` function.

`${{ safeHTML (replace (printf "%.2f" (.Params.custom)) "." "<sup>") }}</sup>`

I'm happy with the result, it's exactly what I wanted..

<b>$85900<sup>00</sup></b>

...except something is missing

## What is missing

Now, I was looking for a way to enhance this function to break thousands. All the prices on my site go between 1,000 and 100,000, and reading such long numbers without a break is difficult.

I want something like this:

<b>$ 85.900<sup>00</sup></b>

_(note: in Spanish, the default thousand separator is a dot, not a comma)_

I've been doing some research, and apparently `printf`relies on Go's `fmt.Sprintf` function, and [there's no support in fmt verbs for thousand separators](https://stackoverflow.com/questions/13020308/how-to-fmt-printf-an-integer-with-thousands-comma).

## How it could be done

Ideally, a function could implement this as easily as javascript's [Number.toLocaleString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString), to save a few replaces in non-english sites.

```
var data = 10000000;
data.toLocaleString('it-IT')
// → 10.000.000
data.toLocaleString('en-US')
// → 10,000,000
With this method, you can also format numbers with currency informations:
data.toLocaleString('it-IT', {style: 'currency', currency: 'EUR'})
// → 10.000.000,00 €
data.toLocaleString('it-IT', {style: 'currency', currency: 'USD'})
// → 10.000.000,00 US$
data.toLocaleString('en-US', {style: 'currency', currency: 'EUR'})
// → €10,000,000.00
data.toLocaleString('en-US', {style: 'currency', currency: 'USD'})
// → $10,000,000.00
```

Note: the examples are taken from [here](https://theghostwhocodes.com/javascript-how-to-format-numbers-to-locale-with-currency-960b49434226?gi=6aef3867d229).

Although I don't really mind replacing the dots and commas myself, and placing the currency symbol, since at the end I can wrap it all in a partial or a shortcode.

Then maybe something as simple as...

`numberFormat (.Params.price)`

And I could rewrite my function as...

`${{ safeHTML (replace (replace (numberFormat (.Params.custom)) "." "<sup>") "," "." ) }}</sup>`

Or something like `numberFormat(number, decimalSeparator, thousandSeparator)` (maybe this extra arguments are optional)

So I could write

`${{ safeHTML (numberFormat .Params.custom "<sup>" "." ) }}</sup>`

Isn't it much better? ♥

## TL;DR

I don't really mind putting the currency symbol myself, or replacing dots with commas, but I _do_ think it's essential for a function to go from `10000.5` to `10,000.50`.

## Disclaimer

If I'm wrong and there's a way of formatting thousands with the current printf function, I blame stack overflow and [this answer](https://stackoverflow.com/questions/13020308/how-to-fmt-printf-an-integer-with-thousands-comma) that says there's no way 😆 No, JK, if there's an alternative I would LOVE to know it!

~ Thanks for reading 💚

**Comment by moorereason** [comment]
This can already be done with `lang.NumFmt` (without localization support).  See https://gohugo.io/functions/numfmt/.

**Comment by jhabdas** [comment]
I did some work with floating point numbers in Hugo to output a medium sized amount of tabular data and found one can combine with [`math` functions](https://gohugo.io/functions/math/), [`replaceRE`](https://gohugo.io/functions/replacere/) when you need to strip leading `-`s and pipe to [`float`](https://gohugo.io/functions/float/) before piping to `lang-NumFmt` which lead to some pretty gnarly templates for just a single locale. Truth be told it's not much easier doing this kind of work in other generators like Jekyll unless wrapping a dep like http://rubymoney.github.io/money/ (or the golang equivalent, if one exists yet).

**Comment by AeonFr** [comment]
Thanks @moorereason  for pointing out the `lang.NumFmt` function, I hadn't find it in the docs nor in Google, but probably because I searched with the wrong terms!

I think the arguments you can pass on to the NumFmt functions are enough for localizate! Specially with the third argument. Passing `"- , ."` uses the 1.000,00 format.


Thanks @jhabdas for the suggestion on using pipes `|`. Code looks much better now, although I was surprised there's not a pipe for the `replace` function and the `NumFmt` pipe doesn't allow for a third argument (If I were to write the number in English, I could pipe this function as well).

To my use case, to avoid the code getting "messy", I created a partial named `price.html`. The partial accepts only a number as an argument, so you call it like:

`{{ partial "price.html" $number }}`

Inside the partial, using the `.` variable points out to the number itself, so I'm passing it as the second param of the `NumFmt` function. The partial looks like this:
 
```
// partials/price.html
${{ replace ( lang.NumFmt 2 (.) "- , ." ) "," "<sup>" | safeHTML }}</sup>
```

Now the output is exactly as I intended.

$12.750<sup>50</sup>

Thanks for the help! 😍
